dev/PushNotifications/PushNotificationManager.cpp

```diff
@@ -221,249 +272,371 @@ namespace winrt::Microsoft::Windows::PushNotifications::implementation
+                    m_pushTriggerRegistration = backgroundTaskRegistration;
+                }
+
+                // Register a sink for the application to receive foreground raw notifications
+                RegisterSinkHelper();
+
+                {
+                    auto lock{ m_lock.lock_exclusive() };
+                    // Register a PushNotificationBackgroundTask to handle background activation scenarios
+                    THROW_IF_FAILED(::CoRegisterClassObject(
+                        m_registeredClsid,
+                        winrt::make<PushNotificationBackgroundTaskFactory>().get(),
+                        CLSCTX_LOCAL_SERVER,
+                        REGCLS_MULTIPLEUSE,
+                        &m_comActivatorRegistration));
+                }
 
-            auto scopeExitToCleanRegistrations{ wil::scope_exit(
-                [&]()
+                scopeExitToCleanRegistrations.release();
+            }
+            else
+            {
                 {
-                    s_comActivatorRegistration.reset();
+                    auto lock{ m_lock.lock_shared() };
+                    THROW_HR_IF(HRESULT_FROM_WIN32(ERROR_ALREADY_REGISTERED), m_singletonForegroundRegistration && m_singletonBackgroundRegistration);
+                }
 
-                    // Clean the task registration only if it was created during this call
-                    if (registeredTaskFromBuilder)
+                auto scopeExitToCleanRegistrations{ wil::scope_exit([&]()
+                {
                     {
-                        registeredTaskFromBuilder.Unregister(true);
+                        auto lock { m_lock.lock_exclusive() };
+
+                        m_comActivatorRegistration.reset();
+                        m_singletonBackgroundRegistration = false;
+                        m_singletonForegroundRegistration = false;
+                        m_registering = false;
                     }
-                }
-            ) };
 
-            if (WI_IsFlagSet(registrationActivators, PushNotificationRegistrationActivators::ComActivator))
-            {
-                GUID taskClsid = details.TaskClsid();
-                THROW_HR_IF(E_INVALIDARG, taskClsid == GUID_NULL);
+                    UnregisterSinkHelper();
+
+                    auto notificationsLongRunningPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
+                    LOG_IF_FAILED(notificationsLongRunningPlatform->UnregisterFullTrustApplication(m_processName.get()));
+                }) };
+
+                auto notificationPlatform{ PushNotificationHelpers::GetNotificationPlatform() };
+                wil::unique_cotaskmem_string unpackagedAppUserModelId;
+                // Apps treated as unpackaged need to call RegisterFullTrustApplication and register with the LRP
+                THROW_IF_FAILED(notificationPlatform->RegisterFullTrustApplication(m_processName.get(), GUID_NULL, &unpackagedAppUserModelId));
```

Comment:
Bug: You are passing GUID_NULL here instead of the GUID activator

Reply:
This sets up a default registration to be overrode by CreateChannelAsync who actually has the remoteId.